utils/rand/rand.go

```diff
@@ -24,8 +29,11 @@ func Uint64() (uint64, error) {
 	return r, nil
 }
 
-// returns a random uint64 strictly less than n
-// errors if n==0
+// Uint64n returns a random uint64 strictly less than `n`.
+// It returns:
+//   - (0, err) if `n==0`
```

Comment:
```suggestion
// Uint64n returns a random uint64 strictly less than `n`. `n` must be greater than 0.
// It returns:
//   - (0, exception) if `n==0`
```

By returning an error here which is not differentiable from the "can't read from system random" exception, we're requiring the caller to crash in this case. Which is OK, it's valid for a function to enforce the contract on its inputs like this, but I think we could also make this (even more) crystal clear in the docs.

I think we should also use the exception terminology for these cases, and separately document the API requirement for non-zero inputs from the enumeration of error cases. 

(This comment also applies to other cases where an an invalid input results in an exception being returned.

Reply:
Yes they are exceptions in this case because they are considered implementation bugs, and code cannot continue to run if the conditions aren't met. I'll clarify the documentation 👌🏼